library/core/src/iter/traits/iterator.rs

```diff
@@ -3493,7 +3493,9 @@ pub trait Iterator {
     ///
     /// Takes each element, adds them together, and returns the result.
     ///
-    /// An empty iterator returns the zero value of the type.
+    /// An empty iterator returns the zero value of the type except in the event
+    /// of summing primitive types that have negative zero values like f32 and f64.
+    /// An empty iterator of these types will return the negative zero value.
```

Comment:
It is fine to add a caution that the "zero" chosen, in the case of a type with two zero values, may not be the one that a programmer was thinking of. However, I believe the wording chosen focuses overly much on "caveat emptor", when the matter is mathematically required to be a certain way:
```suggestion
    /// An empty iterator returns the *additive identity* ("zero") of the type,
    /// which is `0` for integers and `-0.0` for floats.
```